main_app/views.py

```diff
@@ -42,58 +44,88 @@ def get_context_data(self, user_id, **kwargs):
+#     template_name = 'main_page.html'
+#
+#     def get_context_data(self, **kwargs):
+#         client = IGDB(6)
+#         context = super().get_context_data(**kwargs)
+#         current_page = self.request.GET.get('page', 1)
+#         games = client.api_get_games_list(**{key: value for key, value in self.request.GET.items()})
+#         pages_amount = client.api_get_last_pages_amount()
+#         left_pages = list([str(i) for i in range(max(int(current_page) - 3, 1), int(current_page))])
+#         right_pages = list([str(i) for i in range(min(int(current_page) + 1, pages_amount + 1),
+#                                                   min(int(current_page) + 4, pages_amount + 1))])
+#         end = int(right_pages[-1]) if right_pages else pages_amount
+#         context.update({
+#             'games': games,
+#             'search': self.request.GET.get('search', ''),
+#             'platforms': self.request.GET.get('platforms', ''),
+#             'genres': self.request.GET.get('genres', ''),
+#             'ur1': self.request.GET.get('ur1', '0'),
+#             'ur2': self.request.GET.get('ur2', '10'),
+#             'pages_amount': pages_amount,
+#             'current_page': current_page,
+#             'left_pages': left_pages,
+#             'right_pages': right_pages,
+#             'end': end
+#         })
+#         return context
+
+
+class MainPageView(LoginRequiredMixin, ListView):
     template_name = 'main_page.html'
+    model = Game
+    paginate_by = 6
 
     def get_context_data(self, **kwargs):
-        client = IGDB(6)
         context = super().get_context_data(**kwargs)
-        current_page = self.request.GET.get('page', 1)
-        games = client.api_get_games_list(**{key: value for key, value in self.request.GET.items()})
-        pages_amount = client.api_get_last_pages_amount()
-        left_pages = list([str(i) for i in range(max(int(current_page) - 3, 1), int(current_page))])
-        right_pages = list([str(i) for i in range(min(int(current_page) + 1, pages_amount + 1),
-                                                  min(int(current_page) + 4, pages_amount + 1))])
-        end = int(right_pages[-1]) if right_pages else pages_amount
         context.update({
-            'games': games,
             'search': self.request.GET.get('search', ''),
             'platforms': self.request.GET.get('platforms', ''),
             'genres': self.request.GET.get('genres', ''),
             'ur1': self.request.GET.get('ur1', '0'),
             'ur2': self.request.GET.get('ur2', '10'),
-            'pages_amount': pages_amount,
-            'current_page': current_page,
-            'left_pages': left_pages,
-            'right_pages': right_pages,
-            'end': end
         })
         return context
 
 
-class DetailPageView(LoginRequiredMixin, TemplateView):
+class DetailPageView(LoginRequiredMixin, DetailView):
     template_name = 'detail_page.html'
+    queryset = Game.objects.all()
 
```

Comment:
```
queryset = Game.objects.all()
context_object_name = 'game'
```
It's a good practice to give good names for variables that are used in templates:
https://docs.djangoproject.com/en/2.2/topics/class-based-views/generic-display/#making-friendly-template-contexts

After this change you can use `game` instead of unreadable `object` in template, check all other views for making context variables friendly, ListView, DetailView, etc